src/init.cpp

```diff
@@ -1975,6 +1975,10 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
             InitWarning(strprintf(_("More than one onion bind address is provided. Using %s for the automatically created Tor onion service."), bind_addr.ToStringIPPort()));
         }
         StartTorControl(bind_addr);
+        proxyType addrOnion;
+        if (gArgs.GetArg("-onion", "") == "" && !GetProxy(NET_ONION, addrOnion) ) {
+            LogPrintf("Tor: reminder: listenonion is not disabled and an outbound Tor proxy is not defined, that could leak your ip\n");
+        }
```

Comment:
I think just `!GetProxy(NET_ONION, addrOnion)` is enough. If the second expression is `true` then for sure the first one is `true` too.

But more importantly, I think this log message is unnecessary - the fact that a node is running a tor onion service does not mean they want to hide their IP address from the world. I guess `-onlynet=onion` implies that.

And even if we want to log a message - the above logic is flawed because we would not log it if `-onion=` is provided but `-proxy=` is not and thus the node would make outgoing IPv4 connections directly (revealing its IP address).

Reply:
In my view Its the other way around , the first might be false like u noticed and the use wants to leak his ip with -onion? 
Since -onion is always without nameproxy and the user must know that that leaks anyway.
So i thought we should not warn him when he wish for that.

To have no -proxy set is in my view more concerning.
At least with deleting the nameproxy leaking  version by default we had until now
we overall are better off with ur suggestion to delete it at all. 

There is no clean solution, when we say -listenonion by default.
even if we use  `IsReachable(NET_ONION) `we could not skip all cases, since all nets are on by default.

Like i said in an other PR its a mess and we should fix it before the transition to v3.
Not even thinking about seeding and the -onlynet flags, its doxing mess wherever u dig deep.
Its almost often counter intuitive crafted and the reverse of what u define with some *privacy* options.

Reply:
@vasild could u imagine that we define a flag -privacynode=maximum,medium,low.
That is by default on maximum?

i.e. maximum no listenonion if no proxy with nameproxy no seeding no dns no uunp onlynet respected everywhere just v3 no persistent keystore bind only to unrouteable addr if Tor etc.
or low just as is.

That would really make things more easy to craft in the different modules core will split anyway, by not annoying the user with warnings and a clear guideline for devs .to
follow when obeying those setting.